pkg/utils/commandexecutor.go

```diff
@@ -1,18 +1,39 @@
 package utils
 
 import (
+	"context"
 	"os/exec"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
 )
 
 //go:generate mockery --name=CommandExecutor
 
 type CommandExecutor interface {
 	Exec(name string, arg ...string) ([]byte, error)
+	ExecContext(ctx context.Context, name string, arg ...string) ([]byte, error)
 }
 
 type Executor struct{}
 
 func (e Executor) Exec(name string, arg ...string) ([]byte, error) {
 	cmd := exec.Command(name, arg...)
+
+	return cmd.Output()
+}
+
+func (e Executor) ExecContext(ctx context.Context, name string, arg ...string) ([]byte, error) {
+	cmd := exec.Command(name, arg...)
```

Comment:
I think we can use the `exec.CommandContext` here: https://pkg.go.dev/os/exec#CommandContext
I think it basically does what you implemented.
You could have simply:
```
cmd := exec.CommandContext(ctx, name, arg...)

return cmd.Output()
```

Reply:
I'm glad you ask 🤗 

That was my first implementation until I discovered that child processes won't be killed because `CommandContext` itself just kills the spawned process (see https://github.com/golang/go/blob/master/src/os/exec/exec.go#L485).

Reply:
Ok, I see.
Actually, I think you are referring to "grandchildren" (new processes spawned by the command process).
Indeed those are not killed.
If we want to kill them, I propose the next solution:

```
cmd := exec.CommandContext(ctx, name, arg...)
cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
cmd.Cancel = func() error {
	err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL)
	if err != nil {
		log.Printf("Error killing process group: %v", err)
	}
	return err
}

cmd.Output()
```

I don't think we need to manage the `ctx.Done()`. This should be managed by the context itself.
If I'm right and the tests are passing, it should pass hehe

Reply:
Brilliant, done in https://github.com/trento-project/agent/pull/376/commits/59445feedac3ae807203229b52fd638ef6223ef6